Remove matscipy dependency

SebastianM-C opened the pull request:
The `neighbourlist` tests were moved to the MolSimPy package and that package was added as a conditional dependency with Requires.jl as discussed in #13

cortner commented on src/nlist.jl:
```diff
@@ -4,18 +4,9 @@
-end
+# include("MatSciPy.jl")
+using MolSimPy
+using .MatSciPy: neighbourlist
```
why is this line still here? Should this not import from `MolSimPy`

SebastianM-C replied:
yeah, sorry.

cortner commented:
Is there a way to have an if-else version of `@require` so that a warning or error message can be printed when somebody tries to load the ASE neighbourlist?

SebastianM-C commented:
I'm not sure. I'll look into it.

SebastianM-C commented:
I'm not sure it's possible given https://github.com/MikeInnes/Requires.jl/issues/41

cortner commented:
Ok, then if  I can indulge in your patience a little longer, could you try something the following: 
```
function neighbourlist(at::ASEAtoms, cutoff::Float64)::MatSciPy.NeighbourList
   if ASE.hasmatscipy
      return MatSciPy.NeighbourList(at, cutoff)
   else 
      @warn("some warning / error message") 
      return nothing 
   end 
end
```
And  the `@require` statement needs to set the flag. I think that may have to be done via `Ref` if I recall correctly.

SebastianM-C commented:
That's a nice idea. I'll try to do it today / tonight. Sorry for the delay.

On Mon, Dec 9, 2019, 18:40 Christoph Ortner <notifications@github.com>
wrote:

> Ok, then if I can indulge in your patience a little longer, could you try
> something the following:
>
> function neighbourlist(at::ASEAtoms, cutoff::Float64)::MatSciPy.NeighbourList
>    if ASE.hasmatscipy
>       return MatSciPy.NeighbourList(at, cutoff)
>    else
>       @warn("some warning / error message")
>       return nothing
>    end
> end
>
> And the @require statement needs to set the flag. I think that may have
> to be done via Ref if I recall correctly.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaMolSim/ASE.jl/pull/14?email_source=notifications&email_token=AHN4U5MO44FDCUFVI434K6LQXZYHDA5CNFSM4JYKFFAKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEGJ2MCI#issuecomment-563324425>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHN4U5P5MT2CNEWCF7PUUW3QXZYHDANCNFSM4JYKFFAA>
> .
>

SebastianM-C commented:
I moved the `neighbourlist` function to ASE as you suggested and removed it from MolSimPy (to avoid conflicts when both are loaded); See also the last two commits here: https://github.com/JuliaMolSim/MolSimPy.jl/commits/master

I have tested this (ASE.jl) locally on windows and it works (i.e. test pass). If you would like to extend the CI to windows (or use the previously mentioned github actions workflows), let me know.

SebastianM-C commented:
I think what remains to be done is to add CI to MolSimPy.jl (see https://github.com/JuliaMolSim/MolSimPy.jl/issues/1)

cortner commented:
Many thanks - I’ll review and merge tomorrow!

cortner commented:
Very elegant. THank you!

cortner commented:
I ll register a new version later, but I’ll first want to test my other packages
